module/executiondatasync/tracker/storage.go

```diff
@@ -0,0 +1,529 @@
+package tracker
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/hashicorp/go-multierror"
+	"github.com/ipfs/go-cid"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module/blobs"
+)
+
+// badger key prefixes
+const (
+	prefixGlobalState  byte = iota + 1 // global state variables
+	prefixLatestHeight                 // tracks, for each blob, the latest height at which there exists a block whose execution data contains the blob
+	prefixBlobRecord                   // tracks the set of blobs at each height
+)
+
+const (
+	globalStateFulfilledHeight = iota + 1 // latest fulfilled block height
+	globalStatePrunedHeight               // latest pruned block height
+)
+
+const cidsPerBatch = 16 // number of cids to track per batch
+
+func retryOnConflict(db *badger.DB, fn func(txn *badger.Txn) error) error {
+	for {
+		err := db.Update(fn)
+		if errors.Is(err, badger.ErrConflict) {
+			continue
+		}
+		return err
+	}
+}
+
+const globalStateKeyLength = 2
+
+func makeGlobalStateKey(state byte) []byte {
+	globalStateKey := make([]byte, globalStateKeyLength)
+	globalStateKey[0] = prefixGlobalState
+	globalStateKey[1] = state
+	return globalStateKey
+}
+
+const blobRecordKeyLength = 1 + 8 + blobs.CidLength
+
+func makeBlobRecordKey(blockHeight uint64, c cid.Cid) []byte {
+	blobRecordKey := make([]byte, blobRecordKeyLength)
+	blobRecordKey[0] = prefixBlobRecord
+	binary.LittleEndian.PutUint64(blobRecordKey[1:], blockHeight)
+	copy(blobRecordKey[1+8:], c.Bytes())
+	return blobRecordKey
+}
```

Comment:
There is this [makePrefix](https://github.com/onflow/flow-go/blob/master/storage/badger/operation/prefix.go#L99) function in badger operation package, that can be reused. 

Any reason we can't reuse the storage modules?

I understand we store them in different database folder, but I think there are low level functions could be reused, no?